docs/configuring.md

```diff
@@ -0,0 +1,254 @@
+
+Where huge page support is available, the following default sizes apply for the object heap:
+
+- Linux on x86: 2 MB
+- Linux on IBM POWER: 16 MB and 2MB (See Note)
+- Linux on IBM Z: 1 MB
+
+<i class="fa fa-pencil-square-o" aria-hidden="true"></i> Note: IBM POWER9 and later systems support either 16 MB or 2 MB huge pages. However PowerVM LPARs support only 16 MB huge pages, while bare-metal machines support only 2 MB huge pages.
+
+
+Transparent HugePage Support (THP) is an automated mechanism of using huge pages to back virtual memory. On Linux kernels that support THP, it is typically enabled by default with the **madvise** option and can be relied on to provide huge pages as required without any user configuration. To disable THP for your application, use the OpenJ9 [`-XX:-TransparentHugePage`](xxtransparenthugepage.md) option on the command line. To disable THP system-wide, change the **sysfs** boot time defaults with the command `transparent_hugepage=never `. For more information about THP see [Transparent HugePage Support](https://www.kernel.org/doc/html/latest/admin-guide/mm/transhuge.html).
+
+### Windows systems
+
+On Windows systems, large pages are typically 2 MB in size. To use large pages, the VM user must have the Windows **Lock pages in memory** setting enabled in the Local Security Policy. Applications must also be run with Admin privileges in order to use large page memory allocations.
+
+For more information, see the following resources from Microsoft:
+
+- [Large page support](https://docs.microsoft.com/en-gb/windows/win32/memory/large-page-support)
+- [`GetLargePageMinimum` function (`memoryapi.h`)](https://docs.microsoft.com/en-gb/windows/win32/api/memoryapi/nf-memoryapi-getlargepageminimum)
+
+### z/OS systems  
+
+When available, 1 MB pageable pages are the default size for the object heap and the code cache. Other page sizes are available for the object heap, depending on the system architecture as shown in the following table:
+
+| Large page size  | System architecture required              | `-Xlp:codecache` | `-Xlp:objectheap`          |
+|------------------|-------------------------------------------|------------------|----------------------------|
+| 2 GB nonpageable | IBM zEnterprise EC12 processor or later   | Not supported    | Supported (64-bit VM only) |
+| 1 MB nonpageable | System z10 processor or later             | Not supported    | Supported (64-bit VM only) |
+| 1 MB pageable    | IBM zEnterprise EC12 processor or later (see Note) |Supported| Supported                  |
+
+<i class="fa fa-pencil-square-o" aria-hidden="true"></i> **Note:** The Flash Express feature (#0402) helps avoids demoting 1 MB pageable pages to 4 KB pages during periods of high memory usage.
```

Comment:
I am not sure if Flash Express is still relevant nowadays.  Have sent an email to David Hom from RSM team to confirm.

Reply:
I've updated the content with the suggestions/corrections from Dave.